### ISSUE TITLE
[FEATURE] Implement scrollable dock / dock pages

### ISSUE THREAD

**Issue by loxK** [Deletescape-Media/Lawnchair#3491]
### Describe the feature

The dock should be scrollable with multiple (adjustable) pages of icons/folders.

### Why would it be useful to add?

Coming from multiple years of Nova, I miss that feature a lot. I was using it all the time. It is very useful for the apps I use the most, but not enough to be on the screen, or that I just don't want to be on the screen like my banking apps.

### Did the feature exist in Lawnchair v2? (Play Store version)

No

### Example(s)

![test](https://github.com/LawnchairLauncher/lawnchair/assets/136687/eaced6b7-5756-441b-ac9b-49d637a2e04d)

With a single setting: the number of pages the dock shows

**Comment by loxK** [comment]
Up.

**Comment by loxK** [comment]
Please reopen.

**Comment by sarojpanda44** [comment]
I feel this is a valid feature request. Scrollable dock reduces the clutter in the home screen

**Comment by eye-wave** [comment]
Yes! This is such a handy feature, but I've only seen it being used in Nova and nowhere else.

**Comment by howels** [comment]
Required for easy migration from Nova now that project is losing development resources

**Comment by herrbrixter** [comment]
This can be closed. It's a duplicate of #3928

**Comment by loxK** [comment]
Nope https://github.com/LawnchairLauncher/lawnchair/issues/3928 is newer, so, it is indeed the duplicate

**Comment by loxK** [comment]
No, https://github.com/LawnchairLauncher/lawnchair/issues/3928 is newer (january 2024)

**Comment by jonhil** [comment]
Push.
Coming from Nova Launcher (RIP 😭) this feature would be greatly appreciated.

**Comment by bitterroots** [comment]
This would be a very welcome addition.

**Comment by Styphoryte** [comment]
> Push.
> Coming from Nova Launcher (RIP 😭) this feature would be greatly appreciated.

RIP. This is about the only thing missing atm, keeping me from switching from Nova because of that so I'm waiting in hopes of this feature being added someday. Last thing may be the page swipe animations and like nova and some other launcher have is setting a page for Google home feed like you could with Nova, but tbh these days I catch myself never using that anyways so not a huge deal with these but the scrollable dock, I'm kinda surprised it's not been added. I feel like one day this launcher will definitely be better than Nova was the way it's looking.

Also the only way I'm able to still use Nova at all without it buggin out with A15's system gestures is by switching to the 3 button navigation classic mode in settings and then use Iconify with root and LSposed in order to hide the navbar using the first option for the navigation bar settings in that app. It's finicky to hide but once it applies then it works even after a reboot it stays hidden I never see it pop-up after this. NavBarhide module wasn't working for me unfortunately so had to use this workaround I figured out. Then I replaced my system gestures with a nice app called Edge Gestures and nova is working just fine. Sometimes it still randomly bugs out and I have to force stop it but eh, I'll set up a shortcut in this gesture app I just mentioned in order to force close apps quicker when this happens... That will help me until I can finally switch to a new launcher lmao.

**Comment by stom66** [comment]
+1 for this feature.

**Comment by giorodz55** [comment]
I just migrated from Nova, I had too many issues and bugs with my pixel 9 pro xl. This is the only feature that I'm missing. Or could be very useful to declutter and streamline our home screen setups.

**Comment by michael-at-boost** [comment]
+1 for this. 
Nova just added ads in the app drawer so I had to jump ship.
Would be really nice to have 2 pages of dock to be able to ease the transition.

**Comment by Tiritibambix** [comment]
This is the only feature that I'm missing.

**Comment by HAWK-6155** [comment]
+1 from me.

It would be a great addition; for some reason I thought this was already a feature (so I was super confused when I couldn't find the setting).

**Comment by Boradumir** [comment]
+1 This was the number 1 feature why I was using Nova Launcher. I would love to use it here.

**Comment by loxK** [comment]
> +1 This was the number 1 feature why I was using Nova Launcher. I would love to use it here.

In case you missed it, it is planned for v16 which is currently in development : https://github.com/LawnchairLauncher/lawnchair/issues/6442

**Comment by Styphoryte** [comment]
> > +1 This was the number 1 feature why I was using Nova Launcher. I would love to use it here.
> 
> In case you missed it, it is planned for v16 which is currently in development : https://github.com/LawnchairLauncher/lawnchair/issues/6442

So if I tried a dev debug build, could I test this feature once it's been pushed? I'm just wondering, how would I know when it would be? Should I just keep an eye on the actions builds or do I have to go looking in commits?

**Comment by 0xAbdiiii** [comment]
> > > +1 This was the number 1 feature why I was using Nova Launcher. I would love to use it here.
> > 
> > In case you missed it, it is planned for v16 which is currently in development : https://github.com/LawnchairLauncher/lawnchair/issues/6442
> 
> So if I tried a dev debug build, could I test this feature once it's been pushed? I'm just wondering, how would I know when it would be? Should I just keep an eye on the actions builds or do I have to go looking in commits?

they will mark this issue as completed

**Comment by Rinyuu** [comment]
This still being worked on? Definitely my # 1 missing feature at the moment.

**Comment by Tiritibambix** [comment]
After years of waiting, this is happening the day I buy octopi launcher pro 😅

**Comment by howels** [comment]
Perfect. New release looks great and can finally start moving onto lawnchair! 
Dock scrolling left/right is a bit hit and miss, often interpreted as pull-down and tends to "stick" unless the swipe is perfectly aligned left or right but definitely functional.

**Comment by Styphoryte** [comment]
> Perfect. New release looks great and can finally start moving onto lawnchair! 
> Dock scrolling left/right is a bit hit and miss, often interpreted as pull-down and tends to "stick" unless the swipe is perfectly aligned left or right but definitely functional.

Ohhhhhhhhhh letssss go just woke up and bouta test this, hells yeah finally!! Also I'm sure that's easy enough to adjust in future updates as well!!

**Comment by Rinyuu** [comment]
Whoah, finally could switch to Lawnchair today because of this. As mentioned by others, the dock does still often detect the wrong swipes (pulls down top bar quickmenu) and swipe on icons within the dock doesn't work, I guess we'll need some kind of "exclude dock for screen gestures" option for that. But otherwise great progress! 

(Oh, and an option to make dock return to initial position on "pressing home", just like the home screen does, would be sweet)
